src/clp_ffi_js/ir/ClpIrV1Decoder.hpp

```diff
@@ -0,0 +1,102 @@
+#ifndef CLP_FFI_JS_CLPIRV1DECODER_HPP
+#define CLP_FFI_JS_CLPIRV1DECODER_HPP
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include <clp/ir/LogEvent.hpp>
+#include <clp/ir/LogEventDeserializer.hpp>
+#include <clp/ir/types.hpp>
+#include <clp/streaming_compression/zstd/Decompressor.hpp>
+#include <clp/TimestampPattern.hpp>
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+namespace clp_ffi_js {
```

Comment:
For everything under `ir` directory we should add an inner namespace: `clp_ffi_js::ir`